### PR TITLE
[OpenSearch Vector Search] Add Method for approximate_search_query_match_text_with_boolean_filter

### PR DESCRIPTION
**Pull request by freemjstudio**
Replace this entire comment with:
  - **Description:** 
I added a function '_approximate_search_query_match_text_with_boolean_filter' at opensearch_vector_search.py. 
 
  - **Issue:**
 To improve search performance of OpenSearch vector search, it should be possible to compare the similarity with the vector embedding as well as with the text field. I'm developing Chatbot with Langchain based on Opensearch index as a vectorDB now and it helped to improve the search performance. 

  - **Dependencies:** any dependencies required for this change -> NO,
  - **Tag maintainer:** for a quicker response, tag the relevant maintainer (see below),
  - **Twitter handle:** we announce bigger features on Twitter. If your PR gets announced, and you'd like a mention, we'll gladly shout you out!

Please make sure your PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` to check this locally.

See contribution guidelines for more information on how to write/run tests, lint, etc: 
https://github.com/langchain-ai/langchain/blob/master/.github/CONTRIBUTING.md

If you're adding a new integration, please include:
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/extras` directory.

If no one reviews your PR within a few days, please @-mention one of @baskaryan, @eyurtsev, @hwchase17.

### CODE REVIEW

**Comment by freemjstudio** [comment]
@hwchase17 
Hi Team, 

I'm working on it recently and I need to improve the query to OpenSearch asap, I hope you check the commit soon :)

Thanks, 
Minjee Woo (freemjstudio)

**Comment by baskaryan** [comment]
cc @naveentatikonda

File: libs/langchain/langchain/vectorstores/opensearch_vector_search.py [hunk 1]
```diff
@@ -633,13 +658,24 @@ def _raw_similarity_search_with_score(
...
 
             if boolean_filter != {}:
-                search_query = _approximate_search_query_with_boolean_filter(
-                    embedding,
-                    boolean_filter,
-                    k=k,
-                    vector_field=vector_field,
-                    subquery_clause=subquery_clause,
-                )
+                if text_field != {}:
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/opensearch_vector_search.py:661]: the default value for this in L625 is a string but here we're checking that it's not an empty dict — is that intentional / correct?

Comment (baskaryan) [reply libs/langchain/langchain/vectorstores/opensearch_vector_search.py:661]: perhaps would make sense to make default value for text_field None and then here we check if `text_field is not None`

Comment (freemjstudio) [reply libs/langchain/langchain/vectorstores/opensearch_vector_search.py:661]: Oh true I will fix the text_field type as a string

**Comment by freemjstudio** [comment]
@baskaryan 
I fixed the bug in L625. Could you take a look at it? 

Thanks,
Minjee Woo

**Pull request closed without merging**